core/modules/ccontrol/MergingRequester.cc

```diff
@@ -227,11 +227,10 @@ bool MergingRequester::_merge() {
 }
 
 void MergingRequester::_setError(int code, std::string const& msg) {
-    LOGF_INFO("setError");
-    LOGF_INFO("setError %1% %2%" % code % msg);
+    LOGF_INFO("setError: code: %1%, message: %2%" % code % msg);
     std::lock_guard<std::mutex> lock(_errorMutex);
-    _error.code = code;
-    _error.msg = msg;
+    _error.setCode(code);
+    _error.setMsg(msg);
```

Comment:
Ok it seems reasonable as Qserv requirement is to management billion of 
stars, and not billion of errors :-)

On 07/07/2015 07:00 PM, Andy wrote:
>
> In core/modules/ccontrol/MergingRequester.cc 
> <https://github.com/lsst/qserv/pull/99#discussion_r34063185>:
>
> >      std::lock_guard<std::mutex> lock(_errorMutex);
> > -    _error.code = code;
> > -    _error.msg = msg;
> > +    _error.setCode(code);
> > +    _error.setMsg(msg);
>
> Reasonable compiler should be able to optimize that temporary away (if 
> all methods are inline). Even if not the performance penalty should be 
> negligible unless you need to create billions of Error instances.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/lsst/qserv/pull/99/files#r34063185>.
>